Community Toolbox/scripts/utils_CommunityToolboxDraw/utils_CommunityToolboxDraw.gml

```diff
@@ -0,0 +1,266 @@
+    draw_arc_vertices(_x, _y, _radius, _anglefrom, _angleto, _precision, false);
+    if (_outline)
+        draw_vertex(_x, _y);
+    
+    draw_primitive_end();
+}
+
+/// @func draw_chord(x,y,radius,anglefrom,angleto,[outline],[precision])
+/// @desc Draws a chord-like shape (i.e. a circle segment) around the given center, with the given radius and between given angles.
+/// @arg {Real} x               The x coordinate of the chord center.
+/// @arg {Real} y               The y coordinate of the chord center.
+/// @arg {Real} radius          The radius of the chord around the center.
+/// @arg {Real} anglefrom       The starting angle of the chord.
+/// @arg {Real} angleto         The ending angle of the chord.
+/// @arg {Bool} [outline]       Whether only the outline of the chord should be drawn or it should be filled.
+/// @arg {Real} [precision]     The precision of the circle the chord's arc is a part of.
+function draw_chord(_x, _y, _radius, _anglefrom, _angleto, _outline = false, _precision = 24) {
+    // making the initial "draw_arc" call to prepare its static struct
+    static draw_arc_init = draw_arc(0, 0, 0, 0, 0);
+    static normalize_angles = draw_arc.normalize_angles;
+    static draw_arc_vertices = draw_arc.draw_arc_vertices;
+    
+    // normalise the angles for easier processing
+    var _normangles = normalize_angles(_anglefrom, _angleto);
+    _anglefrom = _normangles[0];
+    _angleto = _normangles[1];
+    if (_anglefrom == _angleto)
+        return;
+    
+    // draw the actual circle segment
+    draw_primitive_begin(_outline ? pr_linestrip : pr_trianglefan);
+    draw_arc_vertices(_x, _y, _radius, _anglefrom, _angleto, _precision, _outline);
+    draw_primitive_end();
+}
+
+/// @func draw_arc_range(x,y,r1,r2,anglefrom,angleto,[outline],[precision])
+/// @desc Draws a band-like shape between two arcs around the given center, with the given radii and between given angles.
+/// @arg {Real} x               The x coordinate of the chord center.
+/// @arg {Real} y               The y coordinate of the chord center.
+/// @arg {Real} r1              The first radius to draw the band between.
+/// @arg {Real} r2              The second radius to draw the band between.
+/// @arg {Real} anglefrom       The starting angle of the arcs.
+/// @arg {Real} angleto         The ending angle of the arcs.
+/// @arg {Bool} [outline]       Whether only the outline of the band should be drawn or it should be filled.
+/// @arg {Real} [precision]     The precision of the circles the band is drawn between.
+function draw_arc_range(_x, _y, _r1, _r2, _anglefrom, _angleto, _outline = false, _precision = 24) {
+    // making the initial "draw_arc" call to prepare its static struct
+    static draw_arc_init = draw_arc(0, 0, 0, 0, 0);
+    static normalize_angles = draw_arc.normalize_angles;
+    
+    static draw_arc_range_vertices = method(static_get(draw_arc), function(_x, _y, _r1, _r2, _anglefrom, _angleto, _precision, _outline) {
```

Comment:
What is the reason for having some, but not all, of these helper functions be bound `methods`, while the others are standalone?

Reply:
The reason I bind `method` here is so that other static methods of `draw_arc` are available here (otherwise I'm pretty sure GM would complain about not seeing the static methods). Others don't have nested static method calls, so they remain unbound.